tests/common/mod.rs

```diff
@@ -213,28 +216,35 @@ pub fn fake_install(pkg: &str, ver: &str) {
     }
 
     // Append new package to it.
+
+    let source = if locally_installed {
+        "path+file:///a/b/c"
+    } else {
+        "registry+https://github.com/rust-lang/crates.io-index"
+    };
+
     writeln!(
         &mut File::options()
             .append(true)
             .open(tmp_cargo_home_crates)
             .unwrap(),
-        "\"{pkg} {ver} (registry+https://github.com/rust-lang/crates.io-index)\" = [\"{pkg}\"]",
+        "\"{pkg} {ver} ({source})\" = [\"{pkg}\"]",
     )
```

Comment:
You should be able to inline `source` here, using `source = ...` as a new argument after the format string in the `writeln!` macro call.